src/sink/wavefront.rs

```diff
@@ -285,13 +378,98 @@ impl Sink for Wavefront {
 
 #[cfg(test)]
 mod test {
+    use quickcheck::{QuickCheck, TestResult};
     use super::*;
     use chrono::{TimeZone, Utc};
     use metric::{TagMap, Telemetry};
     use sink::Sink;
     use std::sync::Arc;
 
     #[test]
+    fn test_no_unpadded_gaps() {
+        fn inner(bin_width: u8, ms: Vec<Telemetry>) -> TestResult {
+            if bin_width == 0 {
+                return TestResult::discard();
+            }
+            // TODO
+            //
+            // The fact that I have to set a bucket here suggests that 'padding'
+            // might be sensible for inclusion as an interator on the
+            // Bucket. But! It doesn't have to be that way, if the
+            // Vec<Telemetry> is pre-prepped.
+            let mut bucket = Buckets::new(bin_width as i64);
+            for m in ms.clone() {
+                bucket.add(m);
+            }
+            let mut padding = padding(bucket.into_iter(), bin_width as i64).peekable();
+
+            while let Some(t) = padding.next() {
+                if let Some(next_t) = padding.peek() {
+                    // When we examine the next point in a series there are
+                    // three possibilities:
+                    //
+                    //  1. the points don't hash the same, so we move on
+                    //  2. the points do hash the same:
+                    //     a. if their timestamps are greater than one span
+                    //        apart then they are both zero
+                    //     b. if both points are non-zero they must not be
+                    //        more than one span apart
+                    if t.hash() == next_t.hash() {
+                        let span = (t.timestamp - next_t.timestamp).abs() / (bin_width as i64);
```

Comment:
In this case bin_width :: u8 so rust'll just pad the underlying bytes to hit i64. I'm sure somewhere in [The Reference](https://doc.rust-lang.org/reference/) there's info on exactly how this works but since they split it up into chapters it's become impossible to search. :/